packages/components/src/box-control/input-control.tsx

```diff
@@ -0,0 +1,211 @@
+			case 'left':
+				modifiedSides.push( 'left' );
+				break;
+			case 'right':
+				modifiedSides.push( 'right' );
+				break;
+		}
+	}
+
+	return modifiedSides.filter( ( s ) => allowedSides.has( s ) );
+}
+
+export default function BoxInputControl( {
+	__next40pxDefaultSize,
+	onChange = noop,
+	onFocus = noop,
+	values,
+	selectedUnits,
+	setSelectedUnits,
+	sides,
+	side,
+	...props
+}: BoxControlInputControlProps ) {
+	const defaultValuesToModify = getSidesToModify( side, sides );
+
+	const handleOnFocus = ( event: React.FocusEvent< HTMLInputElement > ) => {
+		onFocus( event, { side } );
+	};
+
+	const handleOnChange = ( nextValues: BoxControlValue ) => {
+		onChange( nextValues );
+	};
+
+	const handleOnValueChange = (
+		next?: string,
+		extra?: { event: React.SyntheticEvent< Element, Event > }
+	) => {
+		const nextValues = { ...values };
+		const isNumeric = next !== undefined && ! isNaN( parseFloat( next ) );
+		const nextValue = isNumeric ? next : undefined;
+		const modifiedSides = getSidesToModify(
+			side,
+			sides,
+			/**
+			 * Supports changing pair sides. For example, holding the ALT key
+			 * when changing the TOP will also update BOTTOM.
+			 */
+			// @ts-expect-error - TODO: event.altKey is only present when the change event was
+			// triggered by a keyboard event. Should this feature be implemented differently so
+			// it also works with drag events?
+			!! extra?.event.altKey
+		);
+
+		modifiedSides.forEach( ( modifiedSide ) => {
+			nextValues[ modifiedSide ] = nextValue;
+		} );
+
+		handleOnChange( nextValues );
+	};
+
+	const handleOnUnitChange = ( next?: string ) => {
+		const newUnits = { ...selectedUnits };
+		defaultValuesToModify.forEach( ( modifiedSide ) => {
+			newUnits[ modifiedSide ] = next;
+		} );
+		setSelectedUnits( newUnits );
+	};
+
+	const mergedValue = getAllValue(
+		values,
+		selectedUnits,
+		defaultValuesToModify
+	);
+	const hasValues = isValuesDefined( values );
+	const isMixed =
+		hasValues &&
+		isValuesMixed( values, selectedUnits, defaultValuesToModify );
+	const mixedPlaceholder = isMixed ? __( 'Mixed' ) : undefined;
+	const [ parsedQuantity, parsedUnit ] =
+		parseQuantityAndUnitFromRawValue( mergedValue );
+	const computedUnit = hasValues
+		? parsedUnit
+		: selectedUnits[ defaultValuesToModify[ 0 ] ];
+	const generatedId = useInstanceId( BoxInputControl, 'box-control-input' );
+	const inputId = [ generatedId, side ].join( '-' );
+
+	return (
+		<InputWrapper key={ `box-control-${ side }` } expanded>
+			<FlexedBoxControlIcon side={ side } sides={ sides } />
+			<Tooltip placement="top-end" text={ LABELS[ side ] }>
+				<StyledUnitControl
+					{ ...props }
+					__shouldNotWarnDeprecated36pxSize
+					__next40pxDefaultSize={ __next40pxDefaultSize }
+					className="component-box-control__unit-control"
+					id={ inputId }
+					isPressEnterToChange
+					value={ mergedValue }
+					onChange={ ( nextValue, extra ) =>
+						handleOnValueChange( nextValue, extra )
+					}
```

Comment:
Nit: looks like we might be able to shorten this:

```suggestion
					onChange={ handleOnValueChange }
```